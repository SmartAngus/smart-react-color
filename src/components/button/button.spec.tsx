import { shallow } from 'enzyme'
import * as React from 'react'
import Button from './index'

test('Button rendered with correct text', () => {
  const clickStub = jest.fn()
  const button = shallow(<Button onClick={clickStub}>Hello World</Button>)

  // Interaction demo
  expect(button.text()).toEqual('Hello World')
  button.simulate('click')
  expect(clickStub).toHaveBeenCalledTimes(1)

  // Snapshot demo
  expect(button).toMatchSnapshot()
})
