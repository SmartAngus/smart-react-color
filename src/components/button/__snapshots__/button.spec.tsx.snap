// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button rendered with correct text 1`] = `
<button
  className="root"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  Hello World
</button>
`;
